assembly/__tests__/runtime/runtime.spec.ts

```diff
@@ -402,6 +415,41 @@ describe("promises", () => {
     // const promise2 = promise.then("contractNameForPromise", "methodName", new Uint8Array(0), 10000000000000);
     // const promise3 = ContractPromise.all([promise2]);
   });
+
+  it("should support contract batch transactions", () => {
+    Context.setPrepaid_gas(10000000000000);
+    expect(context.accountBalance).toBe(u128.from(9))
+    ContractPromiseBatch.create("alice").transfer(u128.from(1))
+    expect(context.accountBalance).toBe(u128.from(8)) // this number is surprising
+  })
+
+  it("should support chained calls", () => {
+    const code = _testBytes();
+
+    // TODO: this sets balance to 14 for some reason, why is that?
+    Context.setAccount_balance(u128.Zero)
+    expect(context.accountBalance).toBe(u128.from(14))
+
+    ContractPromiseBatch
+      .create("app-v1.bob.testnet")
+      .create_account()
+      .transfer(u128.from(1))
+      .add_full_access_key(new Uint8Array(0)) // TODO: how to get signer public key?

```

Comment:
wasn't clear how to get signer public key, are we missing something on the `context` object?  actually just added this but the test failed while trying to set using `Context.setSigner_account_pk()` and i'm not sure how to fix

Reply:
You have to encode it into base58 and pass it as a string. (the glue code handles the conversion to UTF8).

Reply:
i think i'm missing something basic here.

**given** this test

```
ContractPromiseBatch
  .create("app-v1.bob.testnet")
  .create_account()
  .transfer(u128.from(1))
  .add_full_access_key(access_key) // TODO: how to get signer public key?
  .deploy_contract(code)
```

**when** i've previously set access key to this value

```
const access_key = util.stringToBytes(context.senderPublicKey)
```

**then** `as-pect` unit tests pass

but **when** i've previously set the access key to this value
```
const access_key = util.parseFromString<Uint8Array>(base58.encode(util.stringToBytes(context.senderPublicKey)))
```

**then** `as-pect` unit tests fail with this error

```
    [Fail]: ✖ should support chained calls
 [Message]: Cannot parse JSON
   [Stack]: RuntimeError: unreachable
```

am i misunderstanding what you mean by base58 encoding the key first?

practicing my given-when-then @chadoh 👋